gno.land/pkg/gnoweb/static/js/renderer.js

```diff
@@ -7,12 +7,87 @@ function renderUsernames(raw) {
   return raw.replace(/( |\n)@([_a-z0-9]{5,16})/, "$1[@$2](/r/demo/users:$2)");
 }
 
+// Create text nodes for the quotes with class names
+const createQuoteNode = (quote) => {
+  const node = document.createElement("a");
+  node.textContent = quote;
+  node.className = "hljs-string";
+  return node;
+};
+
 function parseContent(source, isCode) {
   if (isCode) {
+    // replace &amp; with & in code blocks
+    source = source.replace(/&amp;/g, "&");
+
```

Comment:
This is an incorrect way to handle the issue. the problem is that parseContent is being given the innerHTML, while it likely should be given the innerText.

In fact, I'm pretty sure this doesn't handle cases like `<` and `>` (see p/demo/avl/node.gno for an example); and it's probably unsafe, too.

Reply:
https://github.com/gnolang/gno/pull/2757/commits/ba13a934348264a5dca3ebdcbe37392d4d2affb5